src/main/webapp/WEB-INF/pages/recentlyViewedProducts.jsp

```diff
@@ -0,0 +1,30 @@
+<c:if test="${not empty sessionScope.recentlyViewedProducts}">
```

Comment:
I guess as another option we can use jsp:useBean with scope of session
but well done that you noticed that!